### ISSUE TITLE
[clippy::doc_markdown] Default whitelist: Add 'InfiniBand' and 'RoCE'

### ISSUE THREAD

**Issue by profetia** [rust-lang/rust-clippy#15557]
### Summary

InfiniBand and RoCE are two common RDMA technologies, and should not be put into `backticks`.

Copy of #10998

### Lint Name

clippy::doc_markdown

### Reproducer

I tried this code:

```rust
#![warn(clippy::doc_markdown)]

//! You need to set `NCCL_IB_GID_INDEX` when working with RoCE.
//! It is not necessary on InfiniBand.
```

I saw this happen:

```
warning: item in documentation is missing backticks
 --> src/main.rs:4:59
  |
4 | //! You need to set `NCCL_IB_GID_INDEX` when working with RoCE.
  |                                                           ^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
note: the lint level is defined here
 --> src/main.rs:2:9
  |
2 | #![warn(clippy::doc_markdown)]
  |         ^^^^^^^^^^^^^^^^^^^^
help: try
  |
4 - //! You need to set `NCCL_IB_GID_INDEX` when working with RoCE.
4 + //! You need to set `NCCL_IB_GID_INDEX` when working with `RoCE`.
  |

warning: item in documentation is missing backticks
 --> src/main.rs:5:28
  |
5 | //! It is not necessary on InfiniBand.
  |                            ^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown
help: try
  |
5 - //! It is not necessary on InfiniBand.
5 + //! It is not necessary on `InfiniBand`.
  |
```

I expected to see this happen:

No warnings

### Version

```text
rustc 1.91.0-nightly (6ba0ce409 2025-08-21)
binary: rustc
commit-hash: 6ba0ce40941eee1ca02e9ba49c791ada5158747a
commit-date: 2025-08-21
host: x86_64-unknown-linux-gnu
release: 1.91.0-nightly
LLVM version: 21.1.0
```
